apps/studio/src/components/referralHero/ReferralDashboard.tsx

```diff
@@ -0,0 +1,287 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { FunctionComponent } from "react";
+import {
+  Box,
+  Divider,
+  IconButton,
+  InputAdornment,
+  Modal,
+  Stack,
+  TextField,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import ContentCopyIcon from "@mui/icons-material/ContentCopy";
+import { useFlags } from "launchdarkly-react-client-sdk";
+import WalletConnectionOverlay from "./WalletConnectionOverlay";
+import ReferralDataBox from "./ReferralDataBox";
+import {
+  getReferralHeroUserCampaignData,
+  useAppDispatch,
+  useAppSelector,
+} from "../../common";
+import {
+  selectUi,
+  setIsConnectWalletModalOpen,
+  setIsReferralDashboardModalOpen,
+  setToastMessage,
+} from "../../modules/ui";
+import {
+  emptyProfile,
+  selectSession,
+  useGetProfileQuery,
+} from "../../modules/session";
+
+interface ReferralDashboardProps {
+  readonly campaignUUID: string;
+}
+
+/**
+ * ReferralDashboard component displays a modal with referral stats and sharable link.
+ * If the user doesn't have a wallet connected, it shows a wallet connection overlay.
+ */
+const ReferralDashboard: FunctionComponent<ReferralDashboardProps> = ({
+  campaignUUID,
+}) => {
+  const { webStudioArtistReferralCampaign } = useFlags();
+  // Check if the referral campaign is enabled before rendering the wrapper component
+  if (!webStudioArtistReferralCampaign) return null;
```

Comment:
Added this to stop the component from loading if the flag is false as this is a wrapper and would load even if the campaign is not active. Probably not what we want.